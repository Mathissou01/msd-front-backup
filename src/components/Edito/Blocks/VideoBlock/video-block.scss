@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.include-media" as im;
@use "src/styles/02-tools/tools.wysiwyg" as wysiwyg;

.c-VideoBlock {
  display: flex;
  flex-direction: column;
  gap: 10px;
  align-items: flex-start;
  width: 100%;

  @include im.media(">=tablet") {
    width: 580px;
    min-height: 328px;
  }

  &__WrapperPlayer {
    position: relative;
    width: 100%;
    height: 100%;
    padding-top: 56.25%;
    overflow: hidden;
    border-radius: 0.7em;
    box-shadow: 0 0 1px transparent;

    @include im.media(">=tablet") {
      width: 580px;
      height: 328px;
    }
  }

  &_hasMaxWith {
    width: 100%;
    max-width: 100%;
  }

  &__Player {
    position: absolute;
    top: 0;
    left: 0;
  }

  &__TranscriptBlock {
    display: flex;
    flex-direction: column;
    gap: 10px;
    align-items: flex-start;
    width: 580px;
  }

  &__Button {
    display: flex;
    gap: 8px;
    align-items: center;
    justify-content: center;
    padding: 12px;
    color: colors.$blue-1;
    font-weight: 500;
    font-size: 14px;
    font-family: DINPro-Medium, sans-serif;
    line-height: 21px;
  }

  &__TranscriptText {
    display: flex;
    gap: 16px;

    @include im.media(">=tablet") {
      padding: 0 0 0 44px;
    }

    &_toggle {
      display: none;
    }
  }

  &__Line {
    @include im.media(">=tablet") {
      min-height: 100%;
      max-height: 100vh;
      border: 2px solid colors.$blue-1;
    }
  }

  &__Text {
    @include wysiwyg.transcript-parse;

    font-size: 16px;
    line-height: 24px;

    @include im.media(">=tablet") {
      font-size: 14px;
      line-height: 21px;
    }
  }

  &__BadUrl {
    padding: 5px;
    color: colors.$error;
    font-size: 18px;
  }
}
