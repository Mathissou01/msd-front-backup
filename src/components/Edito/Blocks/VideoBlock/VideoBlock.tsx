import classNames from "classnames";
import parse from "html-react-parser";
import React, { useState, useEffect } from "react";
import Image from "next/image";
import ReactPlayer from "react-player";
import { makePublicAssetPath } from "../../../../lib/utilities";
import "./video-block.scss";

interface IVideoBlockProps {
  videoLink: string;
  transcriptText?: string;
  hasMaxWith?: boolean;
}

export default function VideoBlock({
  videoLink,
  transcriptText,
  hasMaxWith,
}: IVideoBlockProps) {
  /* Static Data */
  const labelButtons = {
    displayTranscriptText: "Masquer la transcription textuelle de la vidéo",
    hiddenTranscriptText: "Lire la transcription textuelle de la vidéo",
  };
  const errorMessage = "L'URL de la vidéo est invalide";

  /* Methods */
  function onToggleDisplayVideo() {
    setIsDisplayTranscriptVideo(!isDisplayTranscriptVideo);
  }

  /* Local Data */
  const [hasWindow, setHasWindow] = useState<boolean>(false);
  const [isDisplayTranscriptVideo, setIsDisplayTranscriptVideo] =
    useState<boolean>(false);
  const parsedHtml = transcriptText ? parse(transcriptText, {}) : null;

  useEffect(() => {
    if (typeof window !== "undefined") {
      setHasWindow(true);
    }
  }, []);

  const videoBlockClasses = classNames({
    hasMaxWith: hasMaxWith,
  });
  return ReactPlayer.canPlay(videoLink) ? (
    <div className="c-VideoBlock">
      <div
        className={`c-VideoBlock__WrapperPlayer ${
          videoBlockClasses && "c-VideoBlock_hasMaxWith"
        }`}
      >
        {hasWindow && (
          <ReactPlayer
            className="c-VideoBlock__Player"
            url={videoLink}
            width="100%"
            height="100%"
            controls
          />
        )}
      </div>
      {parsedHtml && (
        <div className="c-VideoBlock__TranscriptBlock">
          <button
            className="c-VideoBlock__Button"
            onClick={onToggleDisplayVideo}
          >
            <Image
              src={makePublicAssetPath(
                isDisplayTranscriptVideo
                  ? "/images/pictos/view.svg"
                  : "/images/pictos/view-off.svg",
              )}
              alt="picto oeil"
              width="24"
              height="24"
            />
            {isDisplayTranscriptVideo
              ? labelButtons.displayTranscriptText
              : labelButtons.hiddenTranscriptText}
          </button>
          <div
            className={classNames("c-VideoBlock__TranscriptText", {
              "c-VideoBlock__TranscriptText_toggle": !isDisplayTranscriptVideo,
            })}
          >
            <div className="c-VideoBlock__Line" />
            <div className="c-VideoBlock__Text">{parsedHtml}</div>
          </div>
        </div>
      )}
    </div>
  ) : (
    //TODO A definir par SUEZ
    <span className="c-VideoBlock__BadUrl">{errorMessage}</span>
  );
}
