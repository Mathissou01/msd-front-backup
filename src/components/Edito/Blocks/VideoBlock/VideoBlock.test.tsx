import { render } from "@testing-library/react";
import VideoBlock from "./VideoBlock";

const mock = {
  videoUrl: "monurlyoutube.com",
};
describe("Video Block", () => {
  it("renders", () => {
    const { container } = render(<VideoBlock videoLink={mock.videoUrl} />);
    expect(container).toMatchSnapshot();
  });
});
