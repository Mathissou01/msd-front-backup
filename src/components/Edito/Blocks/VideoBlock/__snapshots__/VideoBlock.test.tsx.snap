// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Block renders 1`] = `
<div>
  <span
    class="c-VideoBlock__BadUrl"
  >
    L'URL de la vid√©o est invalide
  </span>
</div>
`;
